pkg/istio/rbac/v1/provider.go

```diff
@@ -96,15 +101,167 @@ func (p *v1) ConvertAthenzModelIntoIstioRbac(m athenz.Model) []model.Config {
+			svc, path, err := common.ParseAssertionResource(zms.DomainName(athenz.NamespaceToDomain(namespace)), assert)
+			if err != nil {
+				continue
+			}
+			// if svc match with current svc, process it and add it to the rules
+			// note that svc defined on athenz can be a regex, need to match the pattern
+			res, e := regexp.MatchString(svc, svcLabel)
+			if e != nil {
+				log.Errorln("error matching string: ", e.Error())
+				continue
+			}
+			if !res {
+				log.Errorf("athenz svc %s does not match with current svc %s", svc, svcLabel)
+				continue
+			}
+			rule := &v1beta1.Rule{}
+			// form rule.From, must initialize internal source here
+			from_principal := &v1beta1.Rule_From{
+				Source: &v1beta1.Source{},
+			}
+			from_requestPrincipal := &v1beta1.Rule_From{
+				Source: &v1beta1.Source{},
+			}
+			// role name should match zms resource name
+			for _, roleName := range athenzModel.Roles {
+				if roleName == role {
+					// add function to enableOriginJwtSubject, following code assume enableOriginJwtSubject is true by default
+					for _, roleMember := range athenzModel.Members[roleName] {
+						res, err := common.CheckAthenzMemberExpiry(roleMember)
+						if err != nil {
+							log.Errorf("error when checking athenz member expiration date, skipping current member: %s, error: %s", roleMember.MemberName, err)
+							continue
+						}
+						if !res {
+							log.Infoln("member expired, skip adding member to authz policy resource, member: ", roleMember.MemberName)
+							continue
+						}
+						res, err = common.CheckAthenzSystemDisabled(roleMember)
+						if err != nil {
+							log.Errorf("error when checking athenz member system disabled, skipping current member: %s, error: %s", roleMember.MemberName, err)
+							continue
+						}
+						if !res {
+							log.Infoln("member expired, skip adding member to authz policy resource, member: ", roleMember.MemberName)
+							continue
+						}
+
+						spiffeName, err := common.MemberToSpiffe(roleMember)
+						if err != nil {
+							log.Errorln("error converting role name to spiffeName: ", err.Error())
+							continue
+						}
+						from_principal.Source.Principals = append(from_principal.Source.Principals, spiffeName)
+						if p.enableOriginJwtSubject {
+							originJwtName, err := common.MemberToOriginJwtSubject(roleMember)
+							if err != nil {
+								log.Errorln(err.Error())
+								continue
+							}
+							from_requestPrincipal.Source.RequestPrincipals = append(from_requestPrincipal.Source.RequestPrincipals, originJwtName)
+						}
+					}
+					//add role spiffee for role certificate
+					roleSpiffeName, err := common.RoleToSpiffe(string(athenzModel.Name), string(roleName))
+					if err != nil {
+						log.Println("error when convert role to spiffe name: ", err.Error())
+						continue
+					}
+					from_principal.Source.Principals = append(from_principal.Source.Principals, roleSpiffeName)
+				}
```

Comment:
Should we break from the loop here after we found the role to avoid additional iterations?